src/dbnode/encoding/options.go

```diff
@@ -30,7 +30,7 @@ import (
 const (
 	defaultDefaultTimeUnit        = xtime.Second
 	defaultByteFieldDictLRUSize   = 4
-	defaultIStreamReaderSizeM3TSZ = 16
+	defaultIStreamReaderSizeM3TSZ = 8 * 2
```

Comment:
Also FYI I was playing around with this setting since it drives how large a buffer we keep in this stream. Seems like it does have some effects on the flamegraphs where the larger buffer avoids time going to the buffer `fill` as often. But the runtimes still seemed pretty variable so it didn't seem like a change we definitely want to make.

Reply:
Interesting, yeah this might be worth playing around with - potentially in a way that only effects query instead of definitely dbnode as well.